feature: Adding SMS auto-subscribe tutorial (#2)
sinch/sinch-sdk-java-quickstart: tutorials/sms/auto-subscribe-app/README.md

# auto-subscribe application sample

This directory contains sample related to Java SDK tutorials: [auto-subscribe](https://developers.sinch.com/docs/sms/tutorials/sms/tutorials/java-sdk/auto-subscribe)

## Requirements

- JDK 21 or later
- [Maven](https://maven.apache.org/)
- [ngrok](https://ngrok.com/docs)
- [Sinch account](https://dashboard.sinch.com)

## Usage

### Configure application settings

> Application settings is using the SpringBoot configuration file: [`application.yaml`](src/main/resources/application.yaml) file and set:

Reviewer: ```suggestion
Application settings are using the SpringBoot configuration file: [`application.yaml`](src/main/resources/application.yaml) file and set:
```

Author: done

#### Sinch credentials
Located in `credentials` section (*you can find all of the credentials you need on your [Sinch dashboard](https://dashboard.sinch.com)*):
- `project-id`: YOUR_project_id
- `key-id`: YOUR_access_key_id
- `key-secret`: YOUR_access_key_secret

#### Server port
Located in `server` section:

> - port: The port to be used to listen incoming request. <em>Default: 8090</em>

Reviewer: ```suggestion
- port: The port to be used to listen to incoming request. <em>Default: 8090</em>
```

Author: Done

### Starting server locally

> Compile and run the application as server onto you localhost.

Reviewer: ```suggestion
Compile and run the application as a server on your localhost.
```

Author: Done

```bash
mvn spring-boot:run
```

### Use ngrok to forward request to local server

Forwarding request to same `8090` port used above:

*Note: The `8090` value is coming from default config and can be changed (see [Server port](#Server port) configuration section)*

```bash
ngrok http 8090
```

ngrok output will contains output like:
```
ngrok                                                                                                                                                                                                                          (Ctrl+C to quit)

...
Forwarding                    https://0e64-78-117-86-140.ngrok-free.app -> http://localhost:8090

```
The line
```
Forwarding                    https://0e64-78-117-86-140.ngrok-free.app -> http://localhost:8090
```
Contains `https://0e64-78-117-86-140.ngrok-free.app` value.

> This value is having to be used to configure your callback from [Sinch dashboard](https://dashboard.sinch.com/sms/api/services)

Reviewer: Don't get it whats intended here, `intended` or `required`?
```suggestion
This value is intended/required to be used to configure your callback from [Sinch dashboard](https://dashboard.sinch.com/sms/api/services)
```

Reviewer 2: Or even simpler: "will be used"

Author: Done
